File: api/src/main/java/com/br/bludata/api/models/Supplier.java
package com.br.bludata.api.models;

import java.util.Date;

import com.br.bludata.api.models.enums.TypePersonEnum;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "supplier")
@Getter
@Setter
public class Supplier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    public Long id;
    
    @Column(name = "name")
    public String name;

    @Column(name = "typePerson")
    public TypePersonEnum typePerson;

    @Column(name = "documentCnpjCpf")
    public String documentCnpjCpf;

    @Column(name = "documentRg")
    public String documentRg;

    @Column(name = "birthDate")
    public Date birthDate;
    
    @Column(name = "createdAt")
    public Date createdAt;

    @Column(name = "phone")
    private Long phone;

    @ManyToOne
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

}

File: app/src/app/models/supplier/supplier.ts
import { Company } from "../company/company";
import { TypePersonEnum } from "../enums/typeperson.enum";

export class Supplier {
    id: number;
    name: string;
    typePerson: TypePersonEnum;
    documentCnpjCpf: string;
    documentRg: string;
    birthDate: Date;
    createdAt: Date;
    phone: number;
    company: Company;
  
    constructor(
      id: number,
      name: string,
      typePerson: TypePersonEnum,
      documentCnpjCpf: string,
      documentRg: string,
      birthDate: Date,
      createdAt: Date,
      phone: number,
      company: Company
    ) {
      this.id = id;
      this.name = name;
      this.typePerson = typePerson;
      this.documentCnpjCpf = documentCnpjCpf;
      this.documentRg = documentRg;
      this.birthDate = birthDate;
      this.createdAt = createdAt;
      this.phone = phone;
      this.company = company;
    }
  }